### PR TITLE
Support file install type

### PR DESCRIPTION
**Pull request by chrisduong**
Hi,

I've just make change following the comments in PR #113.

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -3,7 +3,19 @@
...
+default['influxdb']['version'] = '0.13.0'
+default['influxdb']['install_type'] = 'package'
+
+default['influxdb']['download_urls'] = {
+  'debian' => 'https://dl.influxdata.com/influxdb/releases',
+  'rhel' => 'https://dl.influxdata.com/influxdb/releases'
+}
+
+# platform_family keyed download sha256 checksums
+default['influxdb']['checksums']['0.10.3-1']['debian'] = '106e4884d8c654f1e43b803b21acc86e6494cafc50a46bcdb4330d0a51bd7aea'
```
Comment (bdangit) [inline attributes/default.rb]: I don't think we should be keeping keys down to the version.  If anything we should just set the shasum to whatever is being set above in `['influxdb']['version']`.  No point in polluting the `node` attributes when it won't be used.

In that sense we should see attributes like this:

```
default['influxdb']['shasums'] = {
  'debian' => 'akjsdhfkjhasdljf...',
  'rhel' => 'asdfjkasdljfh....',
```

File: resources/install.rb [hunk 2]
```diff
@@ -5,42 +5,79 @@
...
-      action :install
+    package 'influxdb' do
+      version node['influxdb']['version']
     end
+  when 'file'
+    if node.platform_family? 'rhel'
+      file_name = "influxdb-#{install_version}.x86_64.rpm"
+      remote_file "#{Chef::Config[:file_cache_path]}/#{file_name}" do
+        source "#{node['influxdb']['download_urls']['rhel']}/#{file_name}"
+        checksum node['influxdb']['checksums'][install_version]['rhel']
```
Comment (bdangit) [inline resources/install.rb]: lets assume that whatever has been predefined, the checksum is used.

**Comment by bdangit** [comment]
A few things:

- [ ] Needs a test-kitchen suite test to make sure if a user a selects `file` and version `0.10.0` or whatever, then the expectation is that the version output by `influxdb version` should match `0.10.0`
- [ ] squash commit if possible.

**Comment by chrisduong** [comment]
The idea for "keyed down for version", which I borrowed from [chef-elasticsearch](https://github.com/elastic/cookbook-elasticsearch/blob/master/attributes/default.rb#L19-L20), is to be look consistent to the version InfluxDB is going to be installed. However, I don't oppose to your simplify resolution.

Those checksums look redundant, but it would make sense if we switch between versions of InfluxDB.

**Comment by chrisduong** [comment]
btw, I've just update test-kitchen suite for `file` and update default recipe to support file install_type.

**Comment by chrisduong** [comment]
Hi @bdangit, its been awhile, could you take a look at my latest comments, see if it make sense. If not, I would revert them.

**Comment by bdangit** [comment]
hi @chrisduong sorry.  I'll check on this today and I'll get back to you.

File: attributes/default.rb (hunk 1 above)
Comment (bdangit) [inline attributes/default.rb]: Can we make this happen?  I know you want to follow elasticsearch as an example, but I also believe in building in whats needed and getting rid of cruft that won't be used.

File: .kitchen.yml [hunk 3]
```diff
@@ -39,6 +39,13 @@ suites:
     run_list:
       - recipe[influxdb-test::default]
     attributes:
+  - name: file
```
Comment (bdangit) [inline .kitchen.yml]: awesome.  If you can, can you make a serverspec test case that tests if your version you selected is actually installed?

You should copy one `config_change` from this https://github.com/bdangit/chef-influxdb/tree/master/test/integration and call it `file`.

And tailor the serverspec tests for this usecase.  Let me know if you need help.

**Comment by chrisduong** [comment]
Hi @bdangit,

Just updated integration test for InfluxDB file installation. Have a nice weekend.

**Comment by bdangit** [comment]
lgtm just tested kitchen locally.

👍

**Comment by bdangit** [comment]
@fishnix what do you think?

**Comment by bdangit** [comment]
alright i'll veto all.  👍  

merging in.

**Comment by bdangit** [comment]
thanks @chrisduong !

**Pull request merged**